### PR TITLE
Add -sFS_DEBUG flag to enable filesystem hooks

### LINKED ISSUE #5076: File changes event listener like `FS.trackingDelegate`
## 🚀 Feature

JS event listener/callback to detect the changes on the Emscripten virtual file system.
For that purpose, [`FS.trackingDelegate`](https://emscripten.org/docs/api_reference/Filesystem-API.html#FS.trackingDelegate%5Bcallback%20name%5D) should be available though `pyodide.FS`, but it is not as of 0.26.2.

```
> pyodide.FS.trackingDelegate
undefined
> pyodide.FS.trackingDelegate['onWriteToFile'] = function(path, bytesWritten) {
...     console.log('Wrote to file "' + path + '" with ' + bytesWritten + ' bytes written');
... };
Uncaught TypeError: Cannot set properties of undefined (setting 'onWriteToFile')
> pyodide.version
'0.26.2'
```

Providing `pyodide.FS.trackingDelegate` is straightforward, and if it's not possible for some reason, it's also helpful to provide any alternative way to do the likes.

### Motivation

I want to make it possible for the JS code to detect the file changes on the file system done by the Python code so that the JS code can read the files reacting to the changes.

### PR DESCRIPTION
**Pull request by ryanking13**
### Description

Resolve #5076

Adds `-sFS_DEBUG` flag to the main module to enable the emscripten filesystem hook (`trackingDelegate`).

### Checklists

- [x] Check the size change of pyodide.asm.js before and after this PR
- [x] Add a [CHANGELOG](https://github.com/pyodide/pyodide/blob/main/docs/project/changelog.md) entry
- [x] Add / update tests

### CODE REVIEW

**Comment by ryanking13** [comment]
Size diff:

__main branch__

```
- pyodide.asm.js:
    Original size: 1201 KB
    Gzip compressed size (level 1): 275 KB
    Gzip compressed size (level 6): 226 KB
    Gzip compressed size (level 9): 226 KB
    Brotli compressed size: 188 KB
```

__this PR__

```
- pyodide.asm.js:
    Original size: 1202 KB
    Gzip compressed size (level 1): 275 KB
    Gzip compressed size (level 6): 226 KB
    Gzip compressed size (level 9): 226 KB
    Brotli compressed size: 188 KB
```

File: src/tests/test_filesystem.py [hunk 1]
```diff
@@ -432,3 +432,31 @@ async def test_nativefs_dup(selenium, runner):
     close(fd2)
     assert file.read() == "hello_read"
     file.close()
+
+
+@pytest.mark.requires_dynamic_linking
```
Comment (hoodmane) [inline src/tests/test_filesystem.py:437]: Why does it require dynamic linking?

**Review by hoodmane — APPROVED** [review]
Thanks. Would be nice to fix the test in the no dylink build unless there is a reason it wouldn't work.

File: src/tests/test_filesystem.py (hunk 1 above)
Comment (ryanking13) [reply src/tests/test_filesystem.py:437]: I forgot to put the same flag in the `ifeq ($(DISABLE_DYLINK), 1)` condition 🙄

Comment (ryanking13) [reply src/tests/test_filesystem.py]: No problem. I don't think it's bad to have duplicate flags if there are only two.

**Comment by ryanking13** [comment]
Thanks for the review!

**Pull request merged**